src/server/scripts/EasternKingdoms/zone_elwynn_forest.cpp

```diff
@@ -414,11 +415,503 @@ class spell_quest_extincteur : public SpellScriptLoader
+            if (ragamuffin2)
+            {
+                ragamuffin2->AI()->Talk(SAY_WOW);
+            }
+        }
+
+        void MoveGeneralHammondClay()
+        {
+            if (GetHammond())
+            {
+                GetHammond()->SetWalk(true);
+                GetHammond()->GetMotionMaster()->MovePoint(0, generalHammondClayPositions[2], true);
+                _events.ScheduleEvent(EVENT_DISMOUNT_HAMMOND_CLAY, 8500);
+            }
+        }
+
+        void SummonAndromath()
+        {
+            TempSummon* andromath = me->SummonCreature(NPC_ANDROMATH, andromathPositions[0]);
+            if (andromath)
+            {
+                _andromathGUID = andromath->GetGUID();
+                andromath->CastSpell(andromath, SPELL_TELEPORT_VISUAL_ONLY_1, true);
+            }
+        }
+
+        void MoveAndromath()
+        {
+            if (GetAndromath())
+            {
+                GetAndromath()->SetWalk(true);
+                GetAndromath()->GetMotionMaster()->MovePoint(0, andromathPositions[2], true);
+            }
+        }
+
+        void SummonDumas()
+        {
+            TempSummon* dumas = me->SummonCreature(NPC_DUMAS, dumasPositions[0]);
+            if (dumas)
+            {
+                _dumasGUID = dumas->GetGUID();
+                dumas->CastSpell(dumas, SPELL_TELEPORT_VISUAL_ONLY_1, true);
+            }
+        }
+
+        void MoveDumas()
+        {
+            if (GetDumas())
+            {
+                GetDumas()->SetWalk(true);
+                GetDumas()->GetMotionMaster()->MovePoint(0, dumasPositions[2], true);
+            }
+        }
+
+        void SummonMinions()
+        {
+            me->CastStop();
+            Talk(SAY_HOGGER_SUMMON_MINIONS);
+            //DoCastSelf(SPELL_SUMMON_MINIONS, true); This works, but the minions just sit there, and then despawn
+            Position hogPos1 = me->GetPosition();
+            Position hogPos2 = me->GetPosition();
+            Position hogPos3 = me->GetPosition();
+            GetPositionWithDistInFront(me, 2.5f, hogPos1);
+            GetPositionWithDistInFront(me, 1.5f, hogPos2);
+            GetPositionWithDistInFront(me, 0.5f, hogPos3);
+
+            float z1 = me->GetMap()->GetHeight(me->GetPhaseShift(), hogPos1.GetPositionX(), hogPos1.GetPositionY(), hogPos1.GetPositionZ());
+            hogPos1.m_positionZ = z1;
+            float z2 = me->GetMap()->GetHeight(me->GetPhaseShift(), hogPos2.GetPositionX(), hogPos2.GetPositionY(), hogPos2.GetPositionZ());
+            hogPos2.m_positionZ = z2;
+            float z3 = me->GetMap()->GetHeight(me->GetPhaseShift(), hogPos3.GetPositionX(), hogPos3.GetPositionY(), hogPos3.GetPositionZ());
+            hogPos3.m_positionZ = z3;
+
+            me->SummonCreature(NPC_HOGGER_MINION, hogPos1);
+            me->SummonCreature(NPC_HOGGER_MINION, hogPos2);
+            me->SummonCreature(NPC_HOGGER_MINION, hogPos3);
```

Comment:
If i understand correctly the only thing changing in above code is the spawn distance, you should use a loop to avoid code repetition

for (float distance : { 0.5f, 1.5f, 2.5f })
{
// Code here
}